### DISCUSSION TITLE
Potential logout improvements

### DISCUSSION THREAD

**Discussion by RVxLab** [filamentphp/filament#1821] [Ideas]
In its current form the current logout route is a GET request, this leaves the endpoint open to a (low-risk) CSRF vulnerability (if you want to class it as such given the context of Filament).

While this can be overwritten in the local route files, I feel this should be implemented in the package directly.

Additionally I think it would be a nice idea to allow users to override the default redirect behaviour after the logout has occured.

## Proposed changes

To address these I propose the following:

1. Change the logout endpoint from GET to POST
2. Put the direct in an action or response which can be bound to an interface like [Fortify does](https://github.com/laravel/fortify/blob/1.x/src/Http/Responses/LogoutResponse.php)

## Pros

Changing the logout endpoint from GET to POST will prevent potential CSRF attacks.

Allowing the user to override the logout action can give more flexibility to the logout route while maintaining the current functionality while not having to require users to override the route itself.

## Cons

Changing the logout endpoint's method could be considered a BC break.

---

I'd like to hear what you think and I would gladly take it upon myself to implement these features.

**Comment by danharrin** [comment]
Sure, a PR would be welcome. I don't count this as a BC break.
